File: src/main/java/net/mjduffin/risk/usecase/UnitStore.java
package net.mjduffin.risk.usecase;

import net.mjduffin.risk.entities.Game;
import net.mjduffin.risk.entities.Player;

import java.util.HashMap;
import java.util.Map;

import static net.mjduffin.risk.entities.Game.State.ALLDRAFT;

public class UnitStore {
    private final int TOTAL_UNITS = 60;
    private final int numPlayers;
    private int units;
    Map<Player, Integer> draftRemaining = new HashMap<>();

    public UnitStore(int numPlayers, Game.State state) {
        this.numPlayers = numPlayers;
        //Initialise draft map
        if (state.equals(ALLDRAFT)) {
            units = TOTAL_UNITS / numPlayers;
        } else {
            units = 10;
        }
    }

    int getUnits() {
        return this.units;
    }

    public void useUnits(int units) {
        this.units -= units;
    }

    //Returns true if a particular player has finished drafting
    public boolean finishedDrafting(Player player) {
        return draftRemaining.get(player) == null || draftRemaining.get(player) == 0;
    }
}

File: risk-lib/src/main/java/net/mjduffin/risk/lib/usecase/UnitStore.kt
package net.mjduffin.risk.lib.usecase

import net.mjduffin.risk.lib.entities.Game
import net.mjduffin.risk.lib.entities.Player

class UnitStore(private val numPlayers: Int, state: Game.State) {
    private val TOTAL_UNITS = 60
    var units = 0
        private set
    var draftRemaining: Map<Player, Int?> = HashMap()
    fun useUnits(units: Int) {
        this.units -= units
    }

    //Returns true if a particular player has finished drafting
    fun finishedDrafting(player: Player): Boolean {
        return draftRemaining[player] == null || draftRemaining[player] == 0
    }

    init {
        //Initialise draft map
        units = if (state == Game.State.ALLDRAFT) {
            TOTAL_UNITS / numPlayers
        } else {
            10
        }
    }
}